File: supervisor-android/app/src/main/java/com/ambe/supervisor/models/Employee.java
package com.ambe.supervisor.models;

public class Employee {
    private String id;
    private String biometricCode;
    private String name;
    private String role;
    private String siteId;
    private String photoUrl;
    private String status;

    public Employee(String id, String biometricCode, String name, String role, String siteId, String photoUrl, String status) {
        this.id = id;
        this.biometricCode = biometricCode;
        this.name = name;
        this.role = role;
        this.siteId = siteId;
        this.photoUrl = photoUrl;
        this.status = status;
    }

    public String getId() { return id; }
    public String getBiometricCode() { return biometricCode; }
    public String getName() { return name; }
    public String getRole() { return role; }
    public String getSiteId() { return siteId; }
    public String getPhotoUrl() { return photoUrl; }
    public String getStatus() { return status; }
}

File: backend/models/Employee.js
const mongoose = require('mongoose');

const EmployeeSchema = new mongoose.Schema({
  id: { type: String, required: true, unique: true },
  biometricCode: { type: String, required: true },
  name: { type: String, required: true },
  role: { type: String, default: 'Janitor' },
  siteId: { type: String, required: true },
  photoUrl: String,
  weeklyOff: { type: String, default: 'Sunday' },
  status: { type: String, enum: ['Active', 'Inactive', 'Pending', 'Deleted'], default: 'Active' },
  joiningDate: String,
  aadharNumber: String,
  panNumber: String,
  bankDetails: {
    accountNumber: String,
    ifscCode: String,
    bankName: String,
    branchName: String
  },
  salaryDetails: {
    baseSalary: Number,
    isDailyRated: Boolean,
    dailyRateOverride: Number,
    deductionBreakdown: {
      advance: Number,
      uniform: Number,
      shoes: Number,
      idCard: Number,
      cbre: Number,
      others: Number
    },
    allowancesBreakdown: {
      travelling: Number,
      others: Number
    },
    basic: Number,
    hra: Number,
    conveyance: Number,
    allowances: Number,
    deductions: Number,
    netSalary: Number,
    paymentType: String
  }
}, { timestamps: true });

module.exports = mongoose.model('Employee', EmployeeSchema);
